### PR TITLE
Move common Linux ptrace definitions to linux_like/mod.rs

### PR DESCRIPTION
**Pull request by mbyzhang**
Move common Linux ptrace definitions to linux_like/mod.rs.

### CODE REVIEW

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Could you check the test failure?
@rustbot author

**Comment by tgross35** [comment]
@mbyzhang are you able to rebase this so we can figure out the CI and get it merged?

**Comment by tgross35** [comment]
@mbyzhang ping, this just needs a rebase

File: src/unix/linux_like/mod.rs [hunk 1]
```diff
@@ -204,6 +204,51 @@ s! {
...
+        pub ret_data: ::__u32,
+    }
+
+    pub struct ptrace_syscall_info {
+        pub op: ::__u8,
+        pub pad: [::__u8; 3],
+        pub arch: ::__u32,
+        pub instruction_pointer: ::__u64,
+        pub stack_pointer: ::__u64,
+        #[cfg(libc_union)]
```
Comment (tgross35) [inline src/unix/linux_like/mod.rs:249]: This `cfg` can be removed since it isn't relevant on `main` anymore

File: src/unix/linux_like/mod.rs [hunk 2]
```diff
@@ -257,6 +302,22 @@ s_no_extra_traits! {
     }
 }
 
+cfg_if! {
+    if #[cfg(libc_union)] {
```
Comment (tgross35) [inline src/unix/linux_like/mod.rs:306]: These types can be moved out of the `cfg` (same reason as above)

**Pull request closed without merging**